### PR TITLE
Makes Notebooks a part of testing (Issue 74)

### PR DESCRIPTION
**Pull request by joshcombes**
This PR adds notebooks to our testing procedure using [`nbval`](https://github.com/computationalmodelling/nbval). See issue https://github.com/rigetti/forest-benchmarking/issues/74 .

The PR includes:
- some small fixes to notebooks that were not working. (The only remaining notebook that does not work is the process tomography, it is documented here https://github.com/rigetti/forest-benchmarking/issues/94 .)
- additions to `conftest.py` to enable skipping slow tests
- additions to `requirements.txt` of `nbval`
- a new test file `test_example_notebooks.py` (notebooks which are longer than 30 seconds are skipped). The test passes if all the cells in a notebook can execute.

### CODE REVIEW

**Comment by joshcombes** [comment]
Thanks to @karalekas fix the code should work locally and remotely.

**Comment by joshcombes** [comment]
This is weird. The tests pass on gitlab but the semaphoreci build fails. @karalekas thoughts?

**Comment by karalekas** [comment]
@joshcombes Two things. (1) Currently the GitLab tests skip the ones that use qvm/quilc (which is a lot of them, so I need to fix that), so there is certainly a chance the GL and Semaphore outputs could be different. (2) Are you able to see the output of the tests on Semaphore? The failure is due to SCS, which is finicky (see #61) and therefore we decided to disable unit tests involving SCS. Take a look at the associated PR to see how we check for the `SKIP_SCS` environment variable and skip tests involving SCS, and if you want you can try to implement that for the distance_measures notebook tests that are failing (I'm happy to do it as well, but might be a good idea to spread the knowledge). Lmk!

**Comment by joshcombes** [comment]
@karalekas the Semaphore thing is hard to decode. In this case it was an easy fix using an `nbval` option.

**Comment by joshcombes** [comment]
3130sc

File: forest/benchmarking/tests/test_example_notebooks.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+
+# ~4 sec
+def test_superoperator_tools_nb():
+    output_str = run_notebook('superoperator_tools.ipynb')
+    print(output_str)
+    assert check_for_failure_and_empty_output(output_str)
+
+
+# Notebook fails at the moment
```
Comment (kylegulshen) [inline forest/benchmarking/tests/test_example_notebooks.py]: Did you ever repeat this error or should we re-incorporate the test?

Comment (joshcombes) [reply forest/benchmarking/tests/test_example_notebooks.py]: I just repeated it. Cell 10 has the error `ValueError: PyQVM can only run run-and-measure style programs: Qubit 0's last operation is a gate` :(

**Pull request merged**